Make on windows - ld crashes?

```
What steps will reproduce the problem?
0. Download optiboot-v5.0a.zip and copy the optiboot contents over the same old 
content in the arduino ide setup. Start cmd.exe and cd to the folder.
machine 1. Use win7 64 bits, omake atmega1284p
machine 2. Use win2003 server, omake atmega1284p

What is the expected output? What do you see instead?
machine 1. There is a window popping up "ld.exe has crashed"...
machine 2. I get "make: *** [optiboot_atmega1284p.elf] Error 1 (and not much 
more of any info why this happened)

What version of the product are you using? On what operating system?
optiboot-v5.0a.zip downloaded from this site. 

Please provide any additional information below.
The zip-file seems to contain some older file than the one in the arduino 1.0.5 
ide when asked about overwriting the files. 
```

Original issue reported on code.google.com by `anders.h...@videocent.com` on 9 Jun 2013 at 11:03

--- a/optiboot/bootloaders/optiboot/Makefile
+++ b/optiboot/bootloaders/optiboot/Makefile
@@ -320,7 +320,7 @@ atmega8: TARGET = atmega8
 atmega8: MCU_TARGET = atmega8
 atmega8: CFLAGS += $(COMMON_OPTIONS)
 atmega8: AVR_FREQ ?= 16000000L 
-atmega8: LDSECTIONS  = -Wl,--section-start=.text=0x1e00 -Wl,--section-start=.version=0x1ffe
+atmega8: LDSECTIONS  = -Wl,--section-start=.text=0x1e00 -Wl,--section-start=.version=0x1ffe -Wl,--gc-sections -Wl,--undefined=optiboot_version
 atmega8: $(PROGRAM)_atmega8.hex
 atmega8: $(PROGRAM)_atmega8.lst
 

--- a/optiboot/bootloaders/optiboot/Makefile.extras
+++ b/optiboot/bootloaders/optiboot/Makefile.extras
@@ -92,7 +92,7 @@ atmega32_isp: isp
 attiny84: TARGET = attiny84
 attiny84: MCU_TARGET = attiny84
 attiny84: CFLAGS += $(COMMON_OPTIONS) -DSOFT_UART -DVIRTUAL_BOOT_PARTITION -Dsave_vect_num=4
-attiny84: LDSECTIONS = -Wl,--section-start=.text=0x1c00 -Wl,--section-start=.version=0x1ffe -Wl,--gc-sections -Wl,--undefined=optiboot_version
+attiny84: LDSECTIONS = -Wl,--section-start=.text=0x1d00 -Wl,--section-start=.version=0x1ffe -Wl,--gc-sections -Wl,--undefined=optiboot_version
 attiny84: $(PROGRAM)_attiny84.hex
 attiny84: $(PROGRAM)_attiny84.lst
 

--- a/optiboot/bootloaders/optiboot/Makefile.extras
+++ b/optiboot/bootloaders/optiboot/Makefile.extras
@@ -18,7 +18,7 @@ atmega88: TARGET = atmega88
 atmega88: MCU_TARGET = atmega88
 atmega88: CFLAGS += $(COMMON_OPTIONS)
 atmega88: AVR_FREQ ?= 16000000L 
-atmega88: LDSECTIONS  = -Wl,--section-start=.text=0x1e00 -Wl,--section-start=.version=0x1ffe
+atmega88: LDSECTIONS  = -Wl,--section-start=.text=0x1e00 -Wl,--section-start=.version=0x1ffe -Wl,--gc-sections -Wl,--undefined=optiboot_version
 atmega88: $(PROGRAM)_atmega88.hex
 atmega88: $(PROGRAM)_atmega88.lst
 
@@ -91,8 +91,8 @@ atmega32_isp: isp
 #
 attiny84: TARGET = attiny84
 attiny84: MCU_TARGET = attiny84
-attiny84: CFLAGS += $(COMMON_OPTIONS) -DSOFT_UART -DVIRTUAL_BOOT_PARTITION
-attiny84: LDSECTIONS = -Wl,--section-start=.text=0x1d00 -Wl,--section-start=.version=0x1ffe
+attiny84: CFLAGS += $(COMMON_OPTIONS) -DSOFT_UART -DVIRTUAL_BOOT_PARTITION -Dsave_vect_num=4
+attiny84: LDSECTIONS = -Wl,--section-start=.text=0x1c00 -Wl,--section-start=.version=0x1ffe -Wl,--gc-sections -Wl,--undefined=optiboot_version
 attiny84: $(PROGRAM)_attiny84.hex
 attiny84: $(PROGRAM)_attiny84.lst
 

